src/sentry/search/events/filter.py

```diff
@@ -764,3 +768,130 @@ def format_search_filter(term, params):
             conditions.append(converted_filter)
 
     return conditions, projects_to_filter, group_ids
+
+
+class QueryFilter(QueryBase):
+    """ Filter logic for a snql query """
+
+    def resolve_where(self, query: str) -> None:
+        try:
+            parsed_terms = parse_search_query(query, allow_boolean=True, params=self.params)
```

Comment:
If this did translate directly to SNQL objects you might have a tricky situation with Issue search which use this function as well.

Reply:
I was thinking of having two subclasses of NodeVisitor. Then different parts of Sentry could choose whether they want the grammar to give them SnQL objects or legacy objects based on the class they use.